src/nvme/util.c

```diff
@@ -419,6 +423,38 @@ void nvme_init_copy_range_f1(struct nvme_copy_range_f1 *copy, __u16 *nlbs,
 	}  
 }
 
+void nvme_init_copy_range_f2(struct nvme_copy_range_f2 *copy, __u32 *snsids,
+			  __u16 *nlbs, __u64 *slbas, __u16 *sopts,
+			  __u32 *eilbrts, __u32 *elbatms, __u32 *elbats,
+			  __u16 nr)
+{
+	int i;
+
+	nvme_init_copy_range((struct nvme_copy_range *)copy, nlbs, slbas,
+		eilbrts, elbatms, elbats, nr);
+
+	for (i = 0; i < nr; i++) {
+		copy[i].snsid = cpu_to_le32(snsids[i]);
+		copy[i].sopt = cpu_to_le16(sopts[i]);
+	}
```

Comment:
It would probably be more performant to set the ranges in a single pass instead of looping over them twice